src/main/java/org/opentripplanner/routing/impl/GraphPathFinder.java

```diff
@@ -188,16 +188,11 @@ public GraphPathFinder(Router router) {
             for (GraphPath path : newPaths) {
                 // path.dump();
                 List<FeedScopedId> tripIds = path.getTrips();
-                List<FeedScopedId> callAndRideTripIds = path.getCallAndRideTrips();
-                for (FeedScopedId tripId : tripIds) {
-                    if (!callAndRideTripIds.contains(tripId)) {
-                        options.banTrip(tripId);
-                    }
-                }
-                if (tripIds.isEmpty()) {
+                if (path.getTrips().isEmpty()) {
                     // This path does not use transit (is entirely on-street). Do not repeatedly find the same one.
                     options.onlyTransitTrips = true;
                 }
+                options.banTripsInPath(path);
```

Comment:
I think this method name would be better as `banTripSequencesInPath` or maybe even `banTripSequencesFoundInPath`

Reply:
Refactored in 6adec27faa2e744f94cd91e4ae4d337a5c25c7c7